internal/functions/download/download.go

```diff
@@ -0,0 +1,87 @@
+package download
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/spf13/afero"
+	"github.com/supabase/cli/internal/utils"
+)
+
+func Run(ctx context.Context, slug string, projectRefArg string, fsys afero.Fs) error {
+	// 1. Sanity checks.
+	projectRef := projectRefArg
+	var scriptDirPath string
+	{
+		if len(projectRefArg) == 0 {
+			ref, err := utils.LoadProjectRef(fsys)
+			if err != nil {
+				return err
+			}
+			projectRef = ref
+		} else if !utils.ProjectRefPattern.MatchString(projectRefArg) {
+			return errors.New("Invalid project ref format. Must be like `abcdefghijklmnopqrst`.")
+		}
+		if err := utils.ValidateFunctionSlug(slug); err != nil {
+			return err
+		}
+	}
+	if err := utils.InstallOrUpgradeDeno(ctx, fsys); err != nil {
+		return err
+	}
+
+	var err error
+	scriptDirPath, err = utils.CopyDenoScripts(ctx, fsys)
+	if err != nil {
+		return err
+	}
+
+	// 2. Download Function.
+	{
+		fmt.Println("Downloading " + utils.Bold(slug))
+		denoPath, err := utils.GetDenoPath()
+		if err != nil {
+			return err
+		}
+
+		resp, err := utils.GetSupabase().GetFunctionBodyWithResponse(ctx, projectRef, slug, func(ctx context.Context, req *http.Request) error {
+			// set encoding header
+			req.Header.Set("Accept-Encoding", "br")
```

Comment:
Possible to put this into the openapi spec for codegen? If not, let's add a todo for now
```suggestion
			// TODO: set encoding header in openapi spec
			req.Header.Set("Accept-Encoding", "br")
```

Reply:
this is actually not needed. Golang client doesn't do uncompress for Brotli, so I will remove this header.